clippy_lints/src/methods/str_splitn.rs

```diff
@@ -209,3 +238,82 @@ fn parse_iter_usage(
         span,
     })
 }
+
+use super::NEEDLESS_SPLITN;
+
+pub(super) fn check_needless_splitn(
+    cx: &LateContext<'_>,
+    method_name: &str,
+    expr: &Expr<'_>,
+    self_arg: &Expr<'_>,
+    pat_arg: &Expr<'_>,
+    count: u128,
+) {
+    if !cx.typeck_results().expr_ty_adjusted(self_arg).peel_refs().is_str() {
+        return;
+    }
+    let ctxt = expr.span.ctxt();
+    let mut app = Applicability::MachineApplicable;
+    if_chain! {
+        if method_name == "splitn";
```

Comment:
Why not rplitn?

Reply:
> Why not rsplitn?

I think the result of rsplitn depends on the string it processes.
For example:
"key=value=add".rsplitn(2, "=").next().unwrap(); is not same as "key=value".rsplitn(2, "=").next().unwrap();

This is why I wanted to deal with rsplitn only for literal at first.

Reply:
But isn't `string.rsplitn(2, ',').next()` the same as `string.rsplit(',').next()`?

Reply:
> But isn't `string.rsplitn(2, ',').next()` the same as `string.rsplit(',').next()`?

I'm very sorry, I was amused by my mistake.

Reply:
Lol! No worries, sometimes brains don't work.